Added Docker to the project

ipantazi opened the pull request:
Added Dockerfile.
Added .env file.
Added docker-compose.yml file.
Added dependency to pom.xml.
Edited application.properties.

ipantazi committed:
Added Dockerfile
Added .env file
Added docker-compose.yml file.
Added dependency to pom.xml.
Edited application.properties.

Elena-Bruyako commented on .env:
```diff
@@ -0,0 +1,10 @@
+MYSQLDB_ROOT_PASSWORD=illya
```
`git rm --cached .env` - .env should never be in version control, add `.env.template` instead with actual values removed - otherwise we would need to search for all env var throughout the project to start it

Elena-Bruyako commented on docker-compose.yml:
```diff
@@ -0,0 +1,36 @@
+        "spring.datasource.password" : "${MYSQLDB_PASSWORD}",
+        "spring.jpa.properties.hibernate.dialect" : "org.hibernate.dialect.MySQL8Dialect"
+      }'
+      JAVA_TOOL_OPTIONS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005"
```
use DEBUG_PORT instead of 5005

Elena-Bruyako commented on src/main/resources/application.properties:
```diff
@@ -1,20 +1,27 @@
+spring.mvc.problemdetails.enabled=true
+spring.jackson.serialization.indent_output=true
+
+#spring.datasource.url=jdbc:mysql://localhost:3306/online_book_store?serverTimezone=UTC
```
let's remove commented code

Elena-Bruyako reviewed (changes requested):
good job! Let's improve your solution

ipantazi committed:
Used DEBUG_PORT instead of 5005 in docker-compose.yml.
Removed commented code in application.properties.
Added .env.template and .gitignore files.
Removed .env file from version control.

liliia-ponomarenko commented on dockerfile:
```diff
@@ -0,0 +1,16 @@
+# Builder stage
```
Dockerfile

liliia-ponomarenko commented on src/main/resources/application.properties:
```diff
@@ -1,20 +1,22 @@
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
-
 spring.jpa.hibernate.ddl-auto=validate
+spring.liquibase.enabled=true
```
It’s  default value

liliia-ponomarenko reviewed (approved):
Good job!

ipantazi committed:
Renamed file Dockerfile.
